src/cascadia/TerminalSettingsModel/Profile.cpp

```diff
@@ -336,3 +338,159 @@ Json::Value Profile::ToJson() const
 
     return json;
 }
+winrt::hstring Profile::EvaluatedIcon()
+{
+    // We cache the result here, so we don't search the path for the exe every time.
+    if (!_evaluatedIcon.has_value())
+    {
+        _evaluatedIcon = _evaluateIcon();
+    }
+    return *_evaluatedIcon;
+}
+
+winrt::hstring Profile::_evaluateIcon() const
+{
+    // If the profile has an icon, return it.
+    if (!Icon().empty())
+    {
+        return Icon();
+    }
+
+    // Otherwise, use NormalizeCommandLine to find the actual exe name. This
+    // will actually search for the exe, including spaces, in the same way that
+    // CreateProcess does.
+    std::wstring cmdline{ NormalizeCommandLine(Commandline().c_str()) };
+    if (cmdline.empty())
+    {
+        return {};
+    }
+
+    // NormalizeCommandLine will return a string with embedded nulls after each
+    // arg. We just want the first one.
+    const size_t beforeNull{ ::wcslen(cmdline.c_str()) };
+    const std::wstring_view exe{ cmdline.c_str(), beforeNull };
+    return winrt::hstring{ exe };

```

Comment:
Just passing `cmdline.c_str()` to the `hstring` constructor will be fine - It has a `hstring(wchar_t const* c)` constructor. With the above comment this turns into:
```cpp
const auto cmdline{ NormalizeCommandLine(Commandline().c_str()) };
// NormalizeCommandLine will return a string with embedded
// nulls after each arg. We just want the first one.
return winrt::hstring{ cmdline.c_str() };
```

Reply:
I disagree generally that we should move from a smart string type to an hstring by using `c_str`. String types know their length; `winrt::hstring(wchar_t*)` must use `wcslen` to re-derive the length. Why do extra work?

Reply:
We only want to return the first segment of the double-null terminated string. (See the comment 😅)